### PR TITLE
docs: update Developer Guide

### PR DESCRIPTION
**Pull request by amresh2000**

### CODE REVIEW

**Review by kianseong — APPROVED** [review]
LGTM!

**Review by wxy1203 — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -168,6 +168,21 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Delete Feature
+
+#### Current Implementation
+
+The `delete` feature is implemented by acting on the current filtered`TaskPanel` with a one-based `Index` specified by the user, getting the target `Task` at the specified index, and removing it from the list.
```
Comment (yilinzyl) [inline docs/DeveloperGuide.md:175]:
```suggestion
The `delete` feature is implemented by acting on the current filtered `TaskPanel` with a one-based `Index` specified by the user, getting the target `Task` at the specified index, and removing it from the list.
```

**Review by yilinzyl — APPROVED** [review]
LGTM for now! Can include a sequence diagram in future iterations

**Pull request merged**